### DISCUSSION TITLE
fine tuning spacy dependency parser for custom data

### DISCUSSION THREAD

**Discussion by zarakolagar120** [explosion/spaCy#8321] [Help: Coding & Implementations] [answered]
Hi, 
I was wondering if it is possible to fine tune the spacy dependency parser for my custom data? if yes, does the dependency parser rely on other components in the processing pipeline (e,g the POS tagger). 
could you please tell me whether the data should have both POS and the dependency annotations in that case?

Thank you very much in advance

**Comment by svlandeg** [comment, marked as answer]
Hi!

You can definitely fine-tune the parser on your custom data. Before you start, it probably makes sense to take a minute to read a little through the docs and understand how spaCy's training system with a config file works: https://spacy.io/usage/training. In particular, you'll want to [source](https://spacy.io/usage/processing-pipelines#sourced-components) the component you want to fine-tune.

The parser produces `token.dep` and `token.head` annotations, and doesn't rely on the tagger (which produces `token.tag`). Often, the two components are trained together though, and you can even share a `tok2vec` architecture between them to allow some information sharing between the two tasks: https://spacy.io/usage/embeddings-transformers#embedding-layers

While you're finetuning your model, make sure it doesn't "forget" patterns it knew before, see also the FAQ: https://github.com/explosion/spaCy/discussions/8226 - "Training an ML model".

Hope that helps you get started!
